src/NuGet.Core/NuGet.Commands/RestoreCommand/RestoreCommand.cs

```diff
@@ -168,10 +168,15 @@ public RestoreCommand(RestoreRequest request)
             _enableNewDependencyResolver = _request.Project.RuntimeGraph.Supports.Count == 0 && ShouldUseNewResolverWithLockFile(_isLockFileEnabled, _request.Project) && !_request.Project.RestoreMetadata.UseLegacyDependencyResolver;
         }
 
-        // Use the new lock file if lock files are not enabled, or if lock files are enabled and .NET 10 SDK is used. Note that the legacy fallback is *false* in this case.
+        // Use the new resolver if lock files are not enabled, or if lock files are enabled and .NET 10 SDK is used. Note that the legacy fallback is *false* in this case.
```

Comment:
@jeffkl I find the "Note that the legacy fallback is false in this case" comment confusing.

Is "legacy" referring to non-SDK style projects, or does it mean "when the new feature is not to be used, this method should return false"?  If it's the latter, I don't understand the point of the comment, because the method name makes it clear that's how it should work.   If "legacy" does refer to non-sdk style projects, then the word "fallback" still confuses me, along with the code not having a special case for non-sdk style projects.

It's out of scope for this PR, but when reading this comment and code for this PR, it gives me a lot of uncertainty that it's working as intended.

Reply:
Legacy is referring to the old resolver.

Legacy fallback is https://github.com/NuGet/NuGet.Client/blob/dev/src/NuGet.Core/NuGet.ProjectModel/ProjectRestoreMetadata.cs#L148, UseLegacyDependencyResolver.

Reply:
Ok, so it basically means "returns true if the new resolver should be used, and returns false if the old resolver should be used".  Considering the method name is `ShouldUseNewResolverWithLockFile`, I'm going to delete that part of the comment, as I think the method name explains it more clearly than the comment does.